Add ext types

sschaef opened the pull request:
See also PRs against [api](https://github.com/msgpack4z/msgpack4z-api/pull/1) and [java07](https://github.com/msgpack4z/msgpack4z-java07/pull/1)

This doesn't include any tests yet. Also, I removed `val ext` in `object MsgpackUnion`, because that seems to be dead code and I couldn't find out what it is supposed to do. The same applies for changes to `abstract class MsgpackUnion`, there are no tests which use that code.

I also updated the version numbers, because I would like to see that you publish a new version. ;)

Fixes #2

sschaef commented:
Checks can't pass since new dependencies are not yet available

xuwei-k commented on src/main/scala/msgpack4z/MsgpackUnion.scala:
```diff
@@ -243,7 +242,15 @@ object MsgpackUnion {
+        val data = new Array[Byte](len)
+        var i = 0
+        while (i < len) {
+          data(i) = unpacker.unpackByte()
```
I think this is incorrent. `unpackByte` only unpack integer types, can not get raw byte.

- https://github.com/msgpack/msgpack-java/blob/0.7.0-M6/msgpack-core/src/main/java/org/msgpack/core/MessageUnpacker.java#L733-L788
- https://github.com/msgpack4z/msgpack4z-java07/blob/v0.1.4/src/main/java/msgpack4z/Msgpack07Unpacker.java#L54-L57
- https://github.com/msgpack4z/msgpack4z-java06/blob/v0.1.1/src/main/java/msgpack4z/Msgpack06Unpacker.java#L73-L76
- https://github.com/msgpack/msgpack-java/blob/msgpack-0.6.12/src/main/java/org/msgpack/unpacker/MessagePackUnpacker.java#L438-L449

sschaef replied:
I finally found the time to look at this again. You were right, the implementation was not correct. I replaced the loop by a call to `readPayload`.

sschaef commented:
I also updated the other PRs against java07 and api. Furthermore, I removed the commits that bumped the version numbers, since - as you said - this can be done automatically.
